lib/spack/llnl/util/lock.py

```diff
@@ -286,7 +288,7 @@ def __repr__(self):
 
     def __str__(self):
         """Readable string (with key fields) of the lock."""
-        location = "{0}[{1}:{2}]".format(self.path, self._start, self._length)
+        location = "{0}[{1}:{2}]".format(str(self.path), self._start, self._length)
```

Comment:
Is this necessary? Using `'{}'.format(Path.cwd())` prints out the path correctly, without needing to call `str()`.

Reply:
I think this was an issue for me on Python3.7

Reply:
Just tried on `3.7.16` and it seems fine! Let's wait until we get it passing CI and then just let CI decide whether we need it.

```sh
> python3.7
Python 3.7.16 (default, Mar 29 2023, 16:15:54) 
[GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pathlib import Path
>>> Path.cwd()
PosixPath('/home/cosmicexplorer/tools/s1')
>>> '{}'.format(Path.cwd())
'/home/cosmicexplorer/tools/s1'
>>> '{}'.format(str(Path.cwd()))
'/home/cosmicexplorer/tools/s1'
```